ISPN-13640 ResilienceMaxRetryIT is failing randomly

danberindei opened the pull request:
https://issues.redhat.com/browse/ISPN-13640

Replacing https://github.com/infinispan/infinispan/pull/9814 and https://github.com/infinispan/infinispan/pull/9821

@diegolovison I opened my own PR in the end to add a high level description in the class javadoc and to also simplify the code a bit.

pruivo commented on server/tests/src/test/java/org/infinispan/server/resilience/ResilienceMaxRetryIT.java:
```diff
@@ -34,78 +41,65 @@
-            // assert that the failed server is the one that we killed
-            assert e.getMessage().contains(killedServer.toString());
+            // Assert that the failed server is the one that we killed
+            assert e.getMessage().contains(stoppedServerAddress.toString()) : e.getMessage();
```
`assertTrue`?

danberindei replied:
Done

pruivo commented on server/tests/src/test/java/org/infinispan/server/resilience/ResilienceMaxRetryIT.java:
```diff
@@ -34,78 +41,65 @@
+      assertEquals(2, currentServers.size());
       for (InetSocketAddress currentServer : currentServers) {
-         if (currentServer.getHostString().equals(killedServer.getHostAddress())) {
-            throw new IllegalStateException("The removed server should not be present in the list");
+         if (currentServer.equals(stoppedServerAddress)) {
+            throw new IllegalStateException(stoppedServerAddress + " should not be present in the list: " + currentServers);
          }
       }
```
`assertFalse(currentServers.contains(stoppedServerAddress));`?

danberindei replied:
I replaced it with a positive assert

pruivo commented on server/tests/src/test/java/org/infinispan/server/resilience/ResilienceMaxRetryIT.java:
```diff
@@ -34,78 +41,65 @@
+            cache.get(ThreadLocalRandom.current().nextInt());
             break;
          } catch (Exception e) {
             // the failure is expected
```
check exception message?

danberindei replied:
Done

pruivo commented on server/testdriver/core/src/main/java/org/infinispan/server/test/core/AbstractInfinispanServerDriver.java:
```diff
@@ -138,11 +139,12 @@ public void prepare(String name) {
 
    @Override
    public void start(String name) {
+      this.name = name;
```
shouldn't be set on `prepare` instead?

danberindei replied:
Done

danberindei commented:
Updated @pruivo!

I also made some changes to the driver so that clients are created with the same host string they'll receive from the servers instead of using `getHostName()` (which may transform 127.0.0.1 into localhost).

pruivo commented:
@danberindei it looks like the authn/authz tests are failing now.

danberindei commented:
I see what went wrong, I changed the "org.infinispan.test.host.address" system property to be `127.0.0.1` instead of `localhost` and that broke the Keycloak config. 
I reverted just that change for now, but if integration tests still fail I'll revert all the driver changes and change the test to compare "resolved" addresses instead of "unresolved" ones.

danberindei commented:
CI looks good, finally!

pruivo commented:
merged! thanks @danberindei !
